File: 2. Loop Problems/Loop problems in python/p15.py
# Python program to find the GCD of two numbers using Euclidean Algorithm


num1 = int(input("Enter the first number: "))
num2 = int(input("Enter the second number: "))


a, b = num1, num2
while b != 0:
    remainder = a % b
    a = b
    b = remainder


print(f"GCD of {num1} and {num2} is {a}")

File: 2. Loop Problems/Loop Problems in c/p15.c
// 



#include <stdio.h>

int main() {
    int num1, num2, gcd, remainder;

    
    printf("Enter two numbers: ");
    scanf("%d %d", &num1, &num2);

    
    int a = num1;
    int b = num2;
    if (a < b) {
        int temp = a;
        a = b;
        b = temp;
    }

    
    while (b != 0) {
        remainder = a % b;
        a = b;
        b = remainder;
    }

    gcd = a; 

    printf("GCD of %d and %d is %d\n", num1, num2, gcd);

    
}
